zrml/swaps/src/lib.rs

```diff
@@ -636,6 +706,11 @@ mod pallet {
         /// * `asset_out`: Asset leaving the pool.
         /// * `asset_amount_out`: Amount that will be transferred from the pool to the provider.
         /// * `max_price`: Market price must be equal or less than the provided value.
+        ///
+        /// # Weight
+        ///
+        /// Complexity: `O(1)` if the scoring rule is CPMM, `O(n)` where `n` is the amount of
+        /// assets if the scoring rule is Rikiddo.
         #[pallet::weight(T::WeightInfo::swap_exact_amount_out_rikiddo(T::MaxAssets::get().into()))]
```

Comment:
I would probably use the path of CPMM here and add a TODO for rikiddo benchmark check.

Reply:
Fixed in 2b420a37a292f1eae806cb9d51cd6eb0f25364e2